SU2_CFD/src/numerics/species/species_diffusion.cpp

```diff
@@ -0,0 +1,83 @@
+/*!
+ * \file turb_diffusion.cpp
+ * \brief Implementation of numerics classes to compute viscous
+ *        fluxes in turbulence problems.
+ * \author T. Kattmann
+ * \version 7.2.0 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2021, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../../include/numerics/species/species_diffusion.hpp"
+
+CAvgGrad_Species::CAvgGrad_Species(unsigned short val_nDim, unsigned short val_nVar, bool correct_grad,
+                                   const CConfig* config)
+    : CAvgGrad_Scalar(val_nDim, val_nVar, correct_grad, config) {}
+
+void CAvgGrad_Species::ExtraADPreaccIn() {}
```

Comment:
inline

Reply:
Still inline if it is non-empty? If these `ExtraADPreaccIn` should be inlined altogether I would push that for all the other instances in a small PR before to stay consistent

Reply:
Well you need to template the class right? so might as well have it all in the hpp